oat-sa/extension-tao-itemqti-pci
# fix/AUT-2551/fix-response-saving-auth-math-entry-pci

**Related to:** [AUT-2551](https://oat-sa.atlassian.net/browse/AUT-2551)

**Description:**
MathEntry Interaction was not able to save a correct response on authoring.
The problem is on a condition that is checking a boolean value, but the property has a default 'false' string value instead of a false boolean value. I tried to change the default value, but it has unexpected consequences, so the best approach and safest is to add an OR condition with this 'false' string, so both cases scenarios are in.
There are 2 scenarios, the first time the item is created with the PCI, and the second is when you edit an existing item with the PCI. In each scenario, the value type is different.

**Changes:**

- Add OR condition on the change listener function.

**How to check:**

- Check capacity to save a correct response on the item creation with a MathEntry Interaction PCI.
- Check capacity to save a correct response on the item edition with a MathEntry Interaction PCI.
- Check data on the `saveItem` XML on saving the item. It must contain the correct result
![image](https://user-images.githubusercontent.com/11692751/189666989-a2845325-f8dc-4d71-aab4-2ffb05205601.png)
- Check output on Preview.
![image](https://user-images.githubusercontent.com/11692751/189667410-26f78cbb-be96-45f6-9c08-c4281c393062.png)

TAE environment on the ticket
https://oat-sa.atlassian.net/browse/AUT-2551?focusedCommentId=198672
